p2p/libp2p/networksharding/factory/sharderFactory.go

```diff
@@ -0,0 +1,64 @@
+package factory
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/libp2p/networksharding"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+var log = logger.GetOrCreate("p2p/networksharding/factory")
+
+type sharderFactory struct {
+	reconnecter        p2p.Reconnecter
+	peerShardResolver  p2p.PeerShardResolver
+	prioBits           uint32
+	pid                peer.ID
+	maxConnectionCount int
+}
+
+// NewSharderFactory creates a new instance of sharderFactory able to create Sharder instances
+//TODO(iulian) improve the parameter passing here
+//TODO(iulian) should not pass reconnecter but a config
+func NewSharderFactory(
+	reconnecter p2p.Reconnecter,
+	peerShardResolver p2p.PeerShardResolver,
+	prioBits uint32,
+	pid peer.ID,
+	maxConnectionCount int,
+) *sharderFactory {
+
+	return &sharderFactory{
+		reconnecter:        reconnecter,
+		peerShardResolver:  peerShardResolver,
+		prioBits:           prioBits,
+		pid:                pid,
+		maxConnectionCount: maxConnectionCount,
+	}
+}
+
+// Create creates new Sharder instances
+//TODO(iulian) make a common interface out of all sharders implementations and replace interface{}
```

Comment:
if TODO is fullfilles there is no more need for the factory.

Reply:
the factory will still be usefull for the create logic encapsulation here, but the return type should not be interface{} but an actual SharderInterface

Reply:
I know, it will be done in a latter PR.